noxfile.py

```diff
@@ -44,13 +44,28 @@
 reset_coverage = True
 
 
+def lock_deps(session: nox.Session):
```

Comment:
```suggestion
def lock_deps(session: nox.Session):
    """Lock all dependency groups into pdm.lock, instead of only specific ones which could result in issues."""
```